common/transport/http/src/rest_api_client.ts

```diff
@@ -152,6 +156,12 @@ export class RestApiClient {
     }
 
     request.end();
+
+    /*Codes_SRS_NODE_IOTHUB_REST_API_CLIENT_18_004: [ `executeApiCall` shall return an `HttpOperation` object. ] */
+    return {
+      /*Codes_SRS_NODE_IOTHUB_REST_API_CLIENT_18_003: [ `HttpOperation.abort` shall call `abort` on the `ClientRequest` object. ] */
+      abort: () => request.abort()
+    };
```

Comment:
I'm really not sold on returning an object here - I do understand why you're doing it: to provide the caller with a way to cancel the request. I'm not sure why we care about cancelling the request instead of letting it die peacefully (making sure we don't get a random error later? or something like that?). 
I wonder what other options we have. return a request id and maintain a list of requests? return the lower-layer http request object directly? 
I'm also concerned by how much context we'll be keeping in memory (because of the closure) if a user decides to hold on to the reference to the returned object.